docs/incubator-proposal.md

```diff
@@ -0,0 +1,194 @@
+# cert-manager proposal
+
+## Problem
+
+Currently in Kubernetes, a number of different components including user
+applications require TLS certificates. Right now, users can provide
+certificates to workloads on the cluster by creating `Secret` resources that
+contain a private/public keypair.
+
+A number of projects (see: [kube-lego](https://github.com/jetstack/kube-lego),
+[kube-cert-manager](https://github.com/PalmStoneGames/kube-cert-manager)
+and [kubernetes-letsencrypt](https://github.com/tazjin/kubernetes-letsencrypt))
+came about in order to automate the retrieval of certificates from an [ACME](https://github.com/ietf-wg-acme/acme/)
+compliant server. This has served the community well, especially with the rise
+in popularity of ACME due to [letsencrypt](https://letsencrypt.org). These
+projects also attempt to renew the certificate as it reaches it's expiry time.
+
+Whilst each project has invented it's own schema/methodology for obtaining
+these certificates, the end goal has largely been the same. To provide a signed
+keypair in a Secret resource to then be consumed, either by an Ingress
+controller or some other workload in-cluster, and then keep that keypair up to
+date.
```

Comment:
Using the term keypair here is a bit clunky, most developer documentation tends to refer to the certificate and it's private key

Reply:
Ack